demisto_sdk/commands/format/format_module.py

```diff
@@ -111,29 +112,40 @@ def format_manager(input: str = None,
 
             if file_type and file_type.value not in UNFORMATTED_FILES:
                 file_type = file_type.value
-                info_res, err_res, skip_res = run_format_on_file(input=file_path,
-                                                                 file_type=file_type,
-                                                                 from_version=from_version,
-                                                                 output=output,
-                                                                 no_validate=no_validate,
-                                                                 verbose=verbose,
-                                                                 update_docker=update_docker,
-                                                                 assume_yes=assume_yes)
-                if err_res:
-                    error_list.append("err_res")
+                info_res, err_res, skip_res, current_content_entity_ids_to_update = run_format_on_file(
+                    input=file_path,
+                    file_type=file_type,
+                    from_version=from_version,
+                    output=output,
+                    no_validate=no_validate,
+                    verbose=verbose,
+                    update_docker=update_docker,
+                    assume_yes=assume_yes
+                )
                 if err_res:
                     log_list.extend([(err_res, print_error)])
                 if info_res:
                     log_list.extend([(info_res, print_success)])
                 if skip_res:
                     log_list.extend([(skip_res, print_warning)])
+                if current_content_entity_ids_to_update:
+                    content_entity_ids_to_update.update(current_content_entity_ids_to_update)
             elif file_type:
                 log_list.append(([f"Ignoring format for {file_path} as {file_type.value} is currently not "
                                   f"supported by format command"], print_warning))
             else:
                 log_list.append(([f"Was unable to identify the file type for the following file: {file_path}"],
                                  print_error))
-
+        if content_entity_ids_to_update:
+            for file in files:
+                file_path = file.replace('\\', '/')
+                with open(file_path, 'r+') as f:
+                    file_content = f.read()
+                    for id_to_replace, updated_id in content_entity_ids_to_update.items():
+                        file_content = file_content.replace(id_to_replace, updated_id)
+                    f.seek(0)
+                    f.write(file_content)
+                    f.truncate()
```

Comment:
maybe we can put this code in the generic_yml and generic_json formatters - I think it would make the entire thing more readable and will require fewer changes like returning `None` in several locations

Reply:
not sure i follow
first need to format all files in order to collect all the updated IDs, and only then it is possible to update the files